### PR TITLE
bugfix: reloaded the etcd configuation data if the index

### LINKED ISSUE #869: bug: Lua variable status may be inconsistent with etcd
In this case, we need to get the full configuration data from etcd.

2019/11/18 15:56:22 [error] 49#49: *726804 [lua] config_etcd. lua:301: failed to fetch data from etcd: The event in requested index is outdated and cleared,  etcd key: /apisix/routes, context: ngx.timer

### PR DESCRIPTION
**Pull request by membphis**
is outdated and cleared.

Fix #869.

### CODE REVIEW

File: lua/apisix/core/config_etcd.lua [hunk 1]
```diff
@@ -194,9 +210,13 @@ local function sync_data(self)
     log.debug("res: ", json.delay_encode(res, true))
     log.debug("headers: ", json.delay_encode(headers, true))
     if not res then
-        if headers then
-          self:upgrade_version(headers["X-Etcd-Index"])
+        if err == "The event in requested index is outdated and cleared" then
```
Comment (moonming) [inline lua/apisix/core/config_etcd.lua]: It will fail when etcd changed this error msg. Is there a better way?

Comment (membphis) [reply lua/apisix/core/config_etcd.lua]: We can check the error messages returned for different etcd versions.

Comment (moonming) [reply lua/apisix/core/config_etcd.lua]: So we need a test case to cover it, otherwise, we can not catch the changes.

Comment (membphis) [reply lua/apisix/core/config_etcd.lua]: It's not easy to test it in APISIX. We can confirm it in lua-resty-etcd.

Comment (membphis) [reply lua/apisix/core/config_etcd.lua]: https://github.com/etcd-io/etcd/blob/a621d807f061e1dd635033a8d6bc261461429e27/Documentation/v2/errorcode.md

it shoud be a static string.

Comment (moonming) [reply lua/apisix/core/config_etcd.lua]: can we throw the error code `401` at the same time?

Comment (membphis) [reply lua/apisix/core/config_etcd.lua]: we checked the error code too.

**Pull request merged**